app/src/main/java/io/apicurio/registry/storage/RegistryStorage.java

```diff
@@ -240,6 +241,15 @@ default boolean isAlive() {
      */
     public SortedSet<Long> getArtifactVersions(String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
 
+    /**
+     * Fetch the versions of the given artifact
+     * @return the artifact versions, limited
+     * @param artifactId the artifact used to fetch versions
+     * @param limit the result size limit
+     * @param offset the number of versions to skip
+     */
+    public VersionSearchResults searchVersions(String artifactId, Integer offset, Integer limit);
```

Comment:
I think to add to the suggestion @alesj made - this signature can use `int` instead of `Integer` and require the caller to provide non-null values.